src/engine/CheckUsePatternTrick.cpp

```diff
@@ -64,11 +64,17 @@ bool isVariableContainedInGraphPatternOperation(
       return ad_utility::contains(arg._inlineValues._variables, variable);
     } else if constexpr (std::is_same_v<T, p::Service>) {
       return ad_utility::contains(arg.visibleVariables_, variable);
+    } else if constexpr (ad_utility::SameAsAny<T, p::PathQuery, p::SpatialQuery,
+                                               p::TextSearchQuery,
+                                               p::NamedCachedResult>) {
+      // For `MagicServiceQuery`s delegate to the nested graph patterns.
+      if (arg.childGraphPattern_.has_value()) {
+        return check(arg.childGraphPattern_.value());
+      }
```

Comment:
I'm not sure if we'd actually want this or if it would be better to always return false because the special service queries are special for a reason.